src/util/verify-chromium-port.js

```diff
@@ -0,0 +1,52 @@
+import { createServer } from 'node:http';
+
+import { UnusablePortError } from '../errors.js';
+
+/**
+ * Determine if a port is available
+ * @param {number} port Port to test
+ * */
+async function isPortAvailable(port) {
+  return new Promise((resolve) => {
+    const server = createServer();
+    server.once('listening', () => {
```

Comment:
For simplicity, you could also consider passing the callback as a parameter to the `listen` function, which is equivalent to manually registering a `listening` listener beforehand.